### PR TITLE
[WIP] Not-found handling for single-resource-provider

### PR DESCRIPTION
**Pull request by dantleech**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no (lets find out!!) |
| Related tickets | #5574 |
| License | MIT |

This PR causes the SingleResourceProvider to throw exception messages when the resource cannot be found, the exception messages indicate how the resolution of the resource was attempted.

### CODE REVIEW

**Comment by dantleech** [comment]
If the tests generally pass and people have no objections I will spec this out.

File: src/Sylius/Bundle/ResourceBundle/Controller/SingleResourceProvider.php [hunk 1]
```diff
@@ -26,14 +26,31 @@ public function get(RequestConfiguration $requestConfiguration, RepositoryInterf
         if (null !== $repositoryMethod = $requestConfiguration->getRepositoryMethod()) {
             $callable = [$repository, $repositoryMethod];
 
+            if (!method_exists($repository, $repositoryMethod)) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/SingleResourceProvider.php]: If we are using an `InvalidArgumentException` anyway, won't it be cleaner to use [webmozart/assert](https://github.com/webmozart/assert)?

Comment (dantleech) [inline src/Sylius/Bundle/ResourceBundle/Controller/SingleResourceProvider.php]: I don't see the added value, but don't mind changing.

Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Controller/SingleResourceProvider.php]: I just find it more readable. Thats all.

**Pull request closed without merging**